cmd/minishift/cmd/start.go

```diff
@@ -409,26 +409,75 @@ func exportContainerImages(hostVm *host.Host, version string) {
 	fmt.Println(fmt.Sprintf("-- Exporting of OpenShift images is occuring in background process with pid %d.", exportCmd.Process.Pid))
 }
 
-// calculateDiskSizeInMB converts a human specified disk size like "1000MB" or "1GB" and converts it into Megabits
-func calculateDiskSizeInMB(humanReadableDiskSize string) int {
-	diskSize, err := units.FromHumanSize(humanReadableDiskSize)
+type sizeValidatorFunc func(name string, diskSize string) error
+type sizeConverterFunc func(input int64) int
+
+func calculateHumanSize(sizeValidator sizeValidatorFunc, name string, humanReadableSize string, unit string, sizeConverter sizeConverterFunc) (int, error) {
+	if stringUtils.HasOnlyNumbers(humanReadableSize) {
```

Comment:
Note: GiB can't be used as a size specifier. It does not pass the validation regex https://github.com/docker/go-units/blob/0dadbb0345b35ec7ef35e228dabb8de89a65bf52/size.go#L34
While internally it is used for displaying purposes:
https://github.com/docker/go-units/blob/0dadbb0345b35ec7ef35e228dabb8de89a65bf52/size.go#L38